### PR TITLE
Move to GitHub Actions for CI and releases

### PR DESCRIPTION
**Pull request by PeterJCLaw**
As part of this we automate the creation of releases now that that's easy to do.

See https://github.com/thread/django-lightweight-queue/pull/86 for what the failures look like.

The publish step is derived from other places I've done this, however it's not been tested in exactly this form.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+          cache-dependency-path: poetry.lock
+
+      - name: Set up Poetry
+        uses: abatilo/actions-poetry@v2
+        with:
+          # Need to use version < 1.2.0 in order to support Python 3.6
```
Comment (judahrand) [inline .github/workflows/check.yml:37]: Ugh 😢 But 3.6 has been EOL for over a year.
Obviously, non-blocking complaint..

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.9"]
+        django-version: ["2.2", "4.0"]
+        exclude:
+          # Python 3.6 is not supported beyond Django 3.2
+          - python-version: "3.6"
+            django-version: "4.0"
```
Comment (judahrand) [inline .github/workflows/check.yml:59]: Can't say that this matrix makes a whole lot of sense to me 🤔 Why not test other Python versions (and Django versions). Just to keep the permutations down?

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+          cache-dependency-path: poetry.lock
```
Comment (judahrand) [inline .github/workflows/check.yml:70]: Why not just `cache: poetry`?

File: .github/workflows/check.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
```
Comment (judahrand) [inline .github/workflows/check.yml:32]: Why not just `cache: poetry`?

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    needs:
+      - run-checks
+    steps:
+      # Using v2 or later strips tag annotations https://github.com/actions/checkout/issues/290
+      - uses: actions/checkout@v1
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.7'
```
Comment (judahrand) [inline .github/workflows/publish.yml:23]: Any reason for `3.7` here?

**Review by judahrand — COMMENTED** [review]
Other than comments/questions (which apply to more places than I've put them in some cases) looks good.

File: .github/workflows/check.yml (hunk 1 above)
Comment (PeterJCLaw) [reply .github/workflows/check.yml:37]: Yeah, my goal here is to make this as close to what was in Circle as possible. At least in terms of behaviour. I'm following this up with a PR to update the supported versions (spoilers at https://github.com/thread/django-lightweight-queue/pull/87).

File: .github/workflows/check.yml (hunk 3 above)
Comment (PeterJCLaw) [reply .github/workflows/check.yml]: This was to do with the install order of python & poetry. Have fixed in https://github.com/thread/django-lightweight-queue/pull/85/commits/ac8524713c693ccce420d768502275e10787a3be by flipping that.

File: .github/workflows/check.yml (hunk 2 above)
Comment (PeterJCLaw) [reply .github/workflows/check.yml:56]: Yeah, we don't really need to lint/type check more than once so this is just for speed.

**Review by judahrand — APPROVED** [review]

**Pull request merged**